src/extras.jl

```diff
@@ -0,0 +1,68 @@
+# Extra stuff
+
+newref!(::Type{Ref{HSA.Region}}, value) = Ref{HSA.Region}(HSA.Region(value))
+
+# Memory
+memset!(ref::Ref{T}, value::T) where T<:Integer = _memset!(ref, value, T)
+
+function _memset!(ref, value, ::Type{T}; len=1) where T
+    ptr = Base.unsafe_convert(Ptr{T}, ref)
+    for i in 1:len
+        unsafe_store!(ptr, value, i)
+    end
+end
+
+function _memset!(ref::Ref{Ptr{Nothing}}, value; len=1)
+    ptr = Base.unsafe_convert(Ptr{UInt8}, ref[])
+    value = UInt8(value)
+    for i in 1:len
+        unsafe_store!(ptr, value, i)
+    end
+end
+    
+# Overloads for interface functions
+
+function getinfo(agent::Agent, attribute::HSA.AgentInfo,
```

Comment:
I'd like if we could allocate the appropriate `Ref` within these `getinfo` functions and just return the resulting unwrapped value.

Reply:
yes--thats exactly what I'm going for. There are a few function "sets" that could be amalgamated and turned into clean multi dispatch functions. With functions like `getinfo()` the only question is _how_ do we want to do the typing. Different attributes have different return types under the umbrella of the void pointer argument. My initial thought was to make a dictionary of values (e.g. `AGENT_INFO_NAME ==> Cstring`) but if you have a better idea I'm open to suggestions.

Reply:
I think a dictionary + `@eval` would be a fine approach.

Reply:
Ok--If it's alright by you I think I will do that in a separate pull request grouping it together with other similar opportunities to condense the API.

Reply:
Sounds good!